python-package/lightgbm/basic.py

```diff
@@ -41,20 +41,26 @@ def info(self, msg: str) -> None:
     def warning(self, msg: str) -> None:
         warnings.warn(msg, stacklevel=3)
 
+    def error(self, msg: str) -> None:
+        warnings.warn(msg, stacklevel=3)
```

Comment:
I guess this was a typo, right?
```suggestion
    def error(self, msg: str) -> None:
        warnings.error(msg, stacklevel=3)
```

Reply:
Seems there's no `error()` in [`warnings`](https://docs.python.org/3/library/warnings.html#available-functions)? I seldom use this module, so correct me if I'm wrong.

Reply:
Ah, indeed! I confused this with `error()` function from logging module. Sorry.
https://docs.python.org/3/library/logging.html#logging.Logger.error

For our dummy logger I think we can just raise an exception then, WDYT?

Reply:
Since this is the default logger, in the future when `error()` is actually needed, we still need to implement it. So I'm inclined to leave a dummy yet usable implementation as default.

Reply:
Why is a `.error()` method being added at all? Per your description in #4783 

> Since only `info` and `warning` are used to log messages, any logger class providing these two methods would suffice.

And I can confirm that such a method isn't used anywhere.

```shelll
git grep -E "\.error"
```

I have a strong preference for not having any "well we might need this in the future" code in the package, as it increases the size of the package and amount of code to be maintained without affecting the user experience. I think this PR should not add a `.error()` implementation at all. @StrikerRUS what do you think?

Reply:
OK, let's remove requirements to have `.error()` method in custom logger class in this PR. But please note that there will be a breaking change in case we'll want to require it in the future.

Reply:
Removed `.error()` related code.

Reply:
Thanks!

> please note that there will be a breaking change in case we'll want to require it in the future

Yes, I understand. I think the case where the `lightgbm` library would want to emit an ERROR-level log but not raise an exception is unlikely to come up.

Even the logger on the C++ side does not have a method for ERROR-level messages: https://github.com/microsoft/LightGBM/blob/17d4e007852b4f0b34b211a674cb26e6751c98e2/include/LightGBM/utils/log.h.